src/EFCore.PG/Internal/NpgsqlOptions.cs

```diff
@@ -43,8 +44,19 @@ public class NpgsqlOptions : INpgsqlOptions
         public virtual bool ReverseNullOrderingEnabled { get; private set; }
 
         /// <inheritdoc />
+        [NotNull]
+        public virtual IReadOnlyList<(string RangeName, Type ElementClrType, string SubTypeName)> RangeMappings { get; private set; }
+
+        /// <inheritdoc />
+        [NotNull]
         public virtual IReadOnlyList<NpgsqlEntityFrameworkPlugin> Plugins { get; private set; }
 
+        public NpgsqlOptions()
+        {
+            RangeMappings = new (string, Type, string)[0];
```

Comment:
We really should have a new type instead that tuple. Otherwise, it looks like copy-paste programming.

Reply:
OK OK I replaced it with `RangeMappingInfo`...